rootfs/docker-entrypoint.sh

```diff
@@ -80,29 +80,47 @@ if [ "$1" = 'postgres' ]; then
 		EOSQL
 		echo
 
+		gosu postgres pg_ctl -D "$PGDATA" -m fast -w stop
+
 		echo
 		for f in /docker-entrypoint-initdb.d/*; do
 			case "$f" in
 				*.sh)  echo "$0: running $f"; . "$f" ;;
-				*.sql)
-					echo "$0: running $f";
-					psql -v ON_ERROR_STOP=1 --username "$POSTGRES_USER" --dbname "$POSTGRES_DB" < "$f"
-					echo
-					;;
 				*)     echo "$0: ignoring $f" ;;
 			esac
 			echo
 		done
 
-		gosu postgres pg_ctl -D "$PGDATA" -m fast -w stop
 		set_listen_addresses '*'
 
 		echo
 		echo 'PostgreSQL init process complete; ready for start up.'
 		echo
 	fi
 
-	exec gosu postgres "$@"
+	# spawn postgres in the background
+	gosu postgres "$@" &
+
+	SERVICE_PID=$!
+
+	# smart shutdown on SIGINT and SIGTERM
+	function on_exit() {
+		kill -TERM $SERVICE_PID
+		wait $SERVICE_PID 2>/dev/null
+		exit 0
+	}
+	trap on_exit INT TERM
+
+	# wait until postgres becomes available
+	until is_running >/dev/null 2>&1; do sleep 1; done
+
+	# Push a fresh backup so we have a new recovery baseline
+	gosu postgres envdir "$WALE_ENVDIR" wal-e backup-push "$PGDATA"
+
+	# Run periodic backups in the background
+	gosu postgres backup &
```

Comment:
Do you think we will need to trap SIGINT and SIGTERM in the backup process as well?